src/saml2/time_util.py

```diff
@@ -241,12 +243,26 @@ def str_to_time(timestr, format=TIME_FORMAT):
     try:
         then = time.strptime(timestr, format)
     except ValueError:  # assume it's a format problem
+        offset = None
         try:
             elem = TIME_FORMAT_WITH_FRAGMENT.match(timestr)
+            if elem:
+                timedata = elem.group(1) + "Z"
+            else:
+                elem = TIME_FORMAT_WITH_FRAGMENT_ZONE.match(timestr)
+                timedata = elem.group(1) + 'Z'
+                offset = timedelta(hours=int(elem.group(4)),
+                                   minutes=int(elem.group(5)))
```

Comment:
Would a mis-formatted string break this code? What if there are no group(4) and group(5)?

Reply:
Yes it will. It that sense it is compatible with the original code that would throw various exceptions when things didn't match. The issue is not that the group is missing, but that the match method can return None.